sql/src/main/java/io/crate/execution/engine/window/ArithmeticOperatorsFactory.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.execution.engine.window;
+
+import io.crate.types.ByteType;
+import io.crate.types.DataType;
+import io.crate.types.DoubleType;
+import io.crate.types.FloatType;
+import io.crate.types.IntegerType;
+import io.crate.types.LongType;
+import io.crate.types.ShortType;
+import io.crate.types.TimestampType;
+
+import java.util.function.BinaryOperator;
+
+class ArithmeticOperatorsFactory {
+
+    private static final BinaryOperator<Double> ADD_DOUBLE_FUNCTION = Double::sum;
+    private static final BinaryOperator<Integer> ADD_INTEGER_FUNCTION = Integer::sum;
+    private static final BinaryOperator<Long> ADD_LONG_FUNCTION = Long::sum;
+    private static final BinaryOperator<Float> ADD_FLOAT_FUNCTION = Float::sum;
+
+    private static final BinaryOperator<Double> SUB_DOUBLE_FUNCTION = (arg0, arg1) -> arg0 - arg1;
+    private static final BinaryOperator<Integer> SUB_INTEGER_FUNCTION = (arg0, arg1) -> arg0 - arg1;
+    private static final BinaryOperator<Long> SUB_LONG_FUNCTION = (arg0, arg1) -> arg0 - arg1;
+    private static final BinaryOperator<Float> SUB_FLOAT_FUNCTION = (arg0, arg1) -> arg0 - arg1;
+
+    static BinaryOperator getAddFunction(DataType dataType) {
```

Comment:
Is it possible to avoid using the raw type variants here? Like:

```suggestion
    static BinaryOperator<?> getAddFunction(DataType<?> dataType) {
```

Reply:
Maybe we can follow up on this and re-use this for the ArithmeticFunctions scalars?

Reply:
oh, I tried but wasn't able to get it working :-| <?> doesn't capture Object for some reason